src/main/java/bio/terra/filesystem/FireStoreFileDao.java

```diff
@@ -64,8 +83,8 @@ public FireStoreFileDao(Firestore firestore, FireStoreUtils fireStoreUtils, Fire
         this.dependencyDao = dependencyDao;
     }
 
-    public UUID createFileStart(FSObject fileToCreate) {
-        if (fileToCreate.getObjectType() != FSObject.FSObjectType.INGESTING_FILE) {
+    public UUID createFileStart(FSFile fileToCreate) {
```

Comment:
It is an invariant that there are no empty directories. When a directory becomes empty on a delete, it is deleted. When a directory is needed, we create it. That is all done (I hope!) within transactions so there is never a time where the visible state violates that invariant. I have documented that invariant in the comments.

I agree that breaking this into parts might make the code a little clearer, but it introduces this state that violates the invariant. I think that would lead to more confusing user-visible behavior.